pkg/agent/config/config_linux.go

```diff
@@ -22,20 +23,36 @@ func applyCRIDockerdAddress(nodeConfig *config.Node) {
 	nodeConfig.CRIDockerd.Address = "unix:///run/k3s/cri-dockerd/cri-dockerd.sock"
 }
 
-func applyContainerdQoSClassConfigFileIfPresent(envInfo *cmds.Agent, nodeConfig *config.Node) {
-	blockioPath := filepath.Join(envInfo.DataDir, "agent", "etc", "containerd", "blockio_config.yaml")
+// applyContainerdQoSClassConfigFileIfPresent checks if QoS class configuration
+// files exist in the filesystem. If so, their paths are included in containerdConfig.
+func applyContainerdQoSClassConfigFileIfPresent(root fs.FS, dataDir string, containerdConfig *config.Containerd) {
+	relaiveDataPath := strings.TrimPrefix(dataDir, string(os.PathSeparator))
+	blockioPath := filepath.Join(relaiveDataPath, "agent", "etc", "containerd", "blockio_config.yaml")
+
+	pathPrefix := ""
+	if filepath.IsAbs(dataDir) {
+		pathPrefix = string(os.PathSeparator)
+	}
 
 	// Set containerd config if file exists
-	if _, err := os.Stat(blockioPath); !errors.Is(err, os.ErrNotExist) {
-		logrus.Infof("BlockIO configuration file found")
-		nodeConfig.Containerd.BlockIOConfig = blockioPath
+	if info, err := fs.Stat(root, blockioPath); !errors.Is(err, fs.ErrNotExist) {
+		if info.IsDir() {
+			logrus.Debugf("Found /%s QoS class configuration file but it is a directory. Skippping.", blockioPath)
+		} else {
+			logrus.Infof("BlockIO configuration file found at /%s.", blockioPath)
+			containerdConfig.BlockIOConfig = filepath.Join(pathPrefix, blockioPath)
```

Comment:
wouldnt be safer and better readability adding the path  (blockio on this case) , already constructed ? instead of assigning it + joining ?  similiar on what u had previous 

something like 
```
whateverfullnamepath := filepath.Join(pathPrefix, blockioPath)

......

 			containerdConfig.BlockIOConfig = whateverfullnamepath

Reply:
See response to other comment.